pkg/apis/pingcap/v1alpha1/tidbmonitor_types.go

```diff
@@ -86,6 +86,15 @@ type PrometheusSpec struct {
 
 	// +optional
 	Ingress *IngressSpec `json:"ingress,omitempty"`
+
+	Container *corev1.Container `json:"container,omitempty"`
```

Comment:
What about using `Containers []*corev1.Container`, then we can also customize the Grafana container as well.

Reply:
Prometheus only have one container ,the grafana container is separate from the Prometheus container.

Reply:
The `containers` field is used for the whole `tidb-monitor` pod. The `prometheus` and `grafana` container are in the same `tidb-monitor` pod.

I recommend using this approach since it is flexible. But it is okay to just use your current way because this pr is already very large. I don't want to add more work to you if you mind. We can totally have another issue to track this

Reply:
Supporting Containers attributes is out of range. This issue is originally to support external configuration.